src/dimension.c

```diff
@@ -450,7 +452,9 @@ dimension_delete_by_hypertable_id(int32 hypertable_id, bool delete_slices)
 				BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(hypertable_id));
 
 	return dimension_scan_internal(scankey, 1, dimension_tuple_delete,
-								   &delete_slices, 0, RowExclusiveLock,
+								   &delete_slices, 0,
+								   DIMENSION_HYPERTABLE_ID_IDX,
```

Comment:
Looking at our `dimension` metadata table I see that we have two indexes on `hypertable_id`: One `(hypertable_id)` and one `(hypertable_id, column_name)`. The latter should suffice. I guess we can fix this in another PR though...

Reply:
Created a PR for this here: https://github.com/timescale/timescaledb/pull/749

Reply:
A good point, but just to be clear, the problem was we were using the hypertable index rather than the dimension id index.

Reply:
Ah sorry, I see this is a different spot; no problem, will look at the PR.

Reply:
yes, I realize that. It's just that I found the unnecessary index while reviewing your fix, which added to the confusion. Figured it is best to clean this up.